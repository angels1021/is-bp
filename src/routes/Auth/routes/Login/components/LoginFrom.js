import React, { PropTypes } from 'react';
import { Field, reduxForm } from 'redux-form/immutable';

import Input from 'common/components/Forms/Input';
import Select from 'common/components/Forms/Select';
import Row from 'common/components/grid/Row';
import Flex from 'common/components/grid/Flex';
import { initialState as initialValues } from 'api/login/constants';

const optionValue = (option) => option && option.value;

const options = [
  {
    value: '/',
    label: 'Register'
  },
  {
    value: '/ms',
    label: 'Management system'
  }
];

const LoginForm = ({ onSubmit, handleSubmit }) => (
  <form
    name="loginForm"
    id="loginForm"
    className="ps-flex column small-12"
    onSubmit={handleSubmit(onSubmit)}
  >
    <Row align="justify" direction="column" className="collapse is-collapse-child" >
      <Flex className="column shrink" direction="column">
        <Row>
          <Field component={Input} type="text" name="username" placeholder="Username" />
        </Row>
        <Row>
          <Field component={Input} type="password" name="password" placeholder="Password" />
        </Row>
        <Row>
          <Field
            component={Select}
            name="location"
            normalize={optionValue}
            options={options}
            plain="true"
          />
        </Row>
        <Field component="input" type="hidden" name="code" />
      </Flex>
      <Flex align="center middle" className=" column shrink ps-flex">
        <button className="button primary">
          <span>
            Submit
          </span>
        </button>
      </Flex>
    </Row>
  </form>
);

LoginForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export default reduxForm({
  form: 'login',
  initialValues
})(LoginForm);
