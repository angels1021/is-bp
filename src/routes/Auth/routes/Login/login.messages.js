export default {
  MODULE: 'auth',
  PAGE: 'login',
  title: {
    defaultMessage: 'login',
    description: 'title for login page'
  },
  submit: {
    defaultMessage: 'Log in',
    description: 'label for login form submit button'
  },
  forgotPassword: {
    defaultMessage: 'forgot my password',
    description: 'label for password recovery link'
  }
};
