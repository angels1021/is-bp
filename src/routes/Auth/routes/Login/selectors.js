/**
 * Login page selectors
 */

import { createSelector } from 'reselect';
import { formValueSelector } from 'redux-form/immutable';

const formSelector = formValueSelector('login');
export const fieldsSelector = (state) => (...keys) => formSelector(state, ...keys);


export const selectField = (...fields) => createSelector(
  fieldsSelector,
  (selector) => selector(...fields)
);

