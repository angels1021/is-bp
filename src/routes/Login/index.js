import React, { Component, PropTypes } from 'react';

export default class Login extends Component {

  render() {
    return (
      <div>
        Im Login
        <div>
          {this.props.children}
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  children: PropTypes.object.isRequired
};
