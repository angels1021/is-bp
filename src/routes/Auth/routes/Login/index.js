/*
* Login page
* */
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
// lang
import { FormattedMessage } from 'react-intl';
import { loginRequest } from 'api/login/actions';
import { selectRequestLoading, selectRequestErrors } from 'common/containers/App/selectors';
import { createRequestId } from 'api/fetchAll/utils';
// global components
import Row from 'common/components/grid/Row';
import Column from 'common/components/grid/Column';
import Flex from 'common/components/grid/Flex';
// local components
import LoginForm from './components/LoginFrom';
// local lang
import messages from './login.messages';

const requestId = createRequestId(`${messages.PAGE}Page`);

export class Login extends Component {

  constructor(props, context) {
    super(props, context);

    this._onSubmit = this._onSubmit.bind(this);
  }

  _onSubmit(formMap, _, formProps) {
    const { valid, pristine } = formProps;
    if (!pristine && valid) {
      const formObj = formMap.toObject();
      this.props.login(formObj);
    }
  }

  render() {
    console.log('loading', this.props.loading(requestId));
    console.log('error', this.props.error(requestId));
    return (
      <Column
        className={`
        small-12
        m-small-6
        medium-4
        large-3
        ps-login-card card`}
      >
        <Flex align="center" className="card-section ps-shrink" >
          <h1 className="text-center" >
            <FormattedMessage {...messages.title} />
          </h1>
        </Flex>
        <Row className="card-section collapse" >
          <LoginForm onSubmit={this._onSubmit} />
        </Row>
      </Column>
    );
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
  loading: PropTypes.func.isRequired,
  error: PropTypes.func.isRequired
};

Login.defaultProps = {
  login: () => {},
  loading: false,
  error: {}
};

// Login.propTypes = {
//   username: PropTypes.string.isRequired,
//   password: PropTypes.string.isRequired,
//   location: PropTypes.string.isRequired,
//   code: PropTypes.string.isRequired
// };
//
const mapStateToProps = createStructuredSelector({
  loading: selectRequestLoading(),
  error: selectRequestErrors()
});

const mapDispatchToProps = (dispatch) => ({
  login: (data) => dispatch(loginRequest(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
