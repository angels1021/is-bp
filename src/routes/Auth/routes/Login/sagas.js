/**
 * Login page sagas
 */
import { fork } from 'redux-saga/effects';
import loginSagas, { NAME } from 'api/auth/sagas.login';
import { callFetchSaga, fetchMessages } from 'api/fetchAll/sagas';
import { PAGE, MODULE } from './login.messages';
import { selectLocale, selectAuth } from '../../selectors';

// act
// fork calling load for each resource so they are simultaneous.
export function* loginFetch() {
  // prep messages
  const messageRequest = [{ module: MODULE, page: PAGE }];
  // call messages
  yield fork(fetchMessages, selectLocale(), messageRequest);
}

// // options to pass to api/fetchAll
// export const fetchOptions = {
//   requestId: `${PAGE}Page`,
//   fetchGenerator: loginFetch,
//   parentId: `${MODULE}Module`,
//   resolveSelector: selectAuth()
// };

export function* fetchFlow() {
  yield fork(callFetchSaga, {
    requestId: `${PAGE}Page`,
    fetchGenerator: loginFetch,
    parentId: `${MODULE}Module`,
    resolveSelector: selectAuth()
  });
}

export default [
  [true, [fetchFlow]], // unnamed, so it will run on every login page load
  [NAME, loginSagas] // names sagas, should only be called once - always name watchers
];
