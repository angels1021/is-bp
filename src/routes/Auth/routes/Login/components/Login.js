/*
 * Login page
 * */
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { SubmissionError } from 'redux-form/immutable';
// lang
import { FormattedMessage } from 'react-intl';
import { loginAction } from 'api/auth/actions';
import { selectRequestLoading, selectRequestErrors } from 'common/containers/App/selectors';
import { createRequestId } from 'api/fetchAll/sagas';
// global components
import Row from 'common/components/grid/Row';
import Column from 'common/components/grid/Column';
import Flex from 'common/components/grid/Flex';
// local components
import LoginForm from './LoginFrom';
// local lang
import messages from '../login.messages';

const requestId = createRequestId(`${messages.PAGE}Page`);

export class Login extends Component {

  constructor(props, context) {
    super(props, context);

    this._onSubmit = this._onSubmit.bind(this);
  }

  _onSubmit(...args) {
    return loginAction(...args)
      .catch((error) => {
        throw new SubmissionError({ form: error.message, _error: 'failed to login' });
      });
  }

  render() {
    console.log('loading', this.props.loading(requestId));
    console.log('error', this.props.error(requestId));
    return (
      <Column
        className={`
        small-12
        m-small-6
        medium-4
        large-3
        ps-login-card card`}
      >
        <Flex align="center" className="card-section ps-shrink" >
          <h1 className="text-center" >
            <FormattedMessage {...messages.title} />
          </h1>
        </Flex>
        <Row className="card-section collapse" >
          <LoginForm onSubmit={this._onSubmit} />
        </Row>
      </Column>
    );
  }
}

Login.propTypes = {
  loading: PropTypes.func.isRequired,
  error: PropTypes.func.isRequired
};

Login.defaultProps = {
  loading: false,
  error: {}
};

// Login.propTypes = {
//   username: PropTypes.string.isRequired,
//   password: PropTypes.string.isRequired,
//   location: PropTypes.string.isRequired,
//   code: PropTypes.string.isRequired
// };
//
const mapStateToProps = createStructuredSelector({
  loading: selectRequestLoading(),
  error: selectRequestErrors()
});

export default connect(mapStateToProps)(Login);
