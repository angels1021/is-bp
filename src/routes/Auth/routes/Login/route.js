import buildRoute from 'utils/routes-helper'; // eslint-disable-line import/no-unresolved
import Login from '../Login';

export default buildRoute({
  path: '/login',
  component(loadModule, injectors) {
    const { fetchDependencies } = injectors;
    Promise.all([
      import('./sagas')
    ]).then(([sagas]) => {
      fetchDependencies(sagas.fetchOptions);
      Login(loadModule);
    }).catch((ex) => {
      console.error('failed to load sagas for login page', ex); // eslint-disable-line no-console
    });
  }
});
