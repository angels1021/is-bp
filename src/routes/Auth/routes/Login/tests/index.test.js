/**
 * Test index
 */
import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../components/Login';

describe('<Login /> container', () => {
  let wrapper;
  const props = {
    login: jest.fn(),
    loading: jest.fn(() => false),
    error: jest.fn(() => {})
  };
  beforeEach(() => {
    wrapper = shallow(<Login {...props} />);
  });
  it('should render correctly', () => {
    // assert
    expect(wrapper).toMatchSnapshot();
  });
  it('should include the <LoginForm />', () => {
    // arrange
    const formWrapper = wrapper.find('ReduxForm');
    // assert
    expect(formWrapper.length).toBe(1);
  });
});
