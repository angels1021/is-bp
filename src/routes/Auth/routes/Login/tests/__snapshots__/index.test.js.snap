// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> container should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Column
    className="
        small-12
        m-small-6
        medium-4
        large-3
        ps-login-card card"
>
    <grid(Flex)
        align="center"
        className="card-section ps-shrink"
        direction=""
    >
        <h1
            className="text-center"
        >
            <FormattedMessage
                defaultMessage="login"
                description="title for login page"
                id="auth.login.title"
                values={Object {}}
            />
        </h1>
    </grid(Flex)>
    <grid(Row)
        align=""
        className="card-section collapse"
        direction=""
    >
        <ReduxForm
            onSubmit={[Function]}
        />
    </grid(Row)>
</Column>,
  "nodes": Array [
    <Column
      className="
        small-12
        m-small-6
        medium-4
        large-3
        ps-login-card card"
>
      <grid(Flex)
            align="center"
            className="card-section ps-shrink"
            direction=""
      >
            <h1
                  className="text-center"
            >
                  <FormattedMessage
                        defaultMessage="login"
                        description="title for login page"
                        id="auth.login.title"
                        values={Object {}}
                  />
            </h1>
      </grid(Flex)>
      <grid(Row)
            align=""
            className="card-section collapse"
            direction=""
      >
            <ReduxForm
                  onSubmit={[Function]}
            />
      </grid(Row)>
</Column>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Login
        error={[Function]}
        loading={[Function]}
        login={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Login {
        "_onSubmit": [Function],
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "error": [Function],
          "loading": [Function],
          "login": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Column
          className="
        small-12
        m-small-6
        medium-4
        large-3
        ps-login-card card"
>
          <grid(Flex)
                    align="center"
                    className="card-section ps-shrink"
                    direction=""
          >
                    <h1
                              className="text-center"
                    >
                              <FormattedMessage
                                        defaultMessage="login"
                                        description="title for login page"
                                        id="auth.login.title"
                                        values={Object {}}
                              />
                    </h1>
          </grid(Flex)>
          <grid(Row)
                    align=""
                    className="card-section collapse"
                    direction=""
          >
                    <ReduxForm
                              onSubmit={[Function]}
                    />
          </grid(Row)>
</Column>,
        "_debugID": 2,
        "_renderedOutput": <Column
          className="
        small-12
        m-small-6
        medium-4
        large-3
        ps-login-card card"
>
          <grid(Flex)
                    align="center"
                    className="card-section ps-shrink"
                    direction=""
          >
                    <h1
                              className="text-center"
                    >
                              <FormattedMessage
                                        defaultMessage="login"
                                        description="title for login page"
                                        id="auth.login.title"
                                        values={Object {}}
                              />
                    </h1>
          </grid(Flex)>
          <grid(Row)
                    align=""
                    className="card-section collapse"
                    direction=""
          >
                    <ReduxForm
                              onSubmit={[Function]}
                    />
          </grid(Row)>
</Column>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Login
    error={[Function]}
    loading={[Function]}
    login={[Function]}
/>,
}
`;
